src/st/cen/11280.c

```diff
@@ -3,274 +3,7 @@ INCLUDE_ASM("st/cen/nonmatchings/11280", TestCollisions);
-            ((arg0 >= g_LayoutObjVertical->posY) ||
-             !(g_LayoutObjVertical->posY != 0xFFFE))) {
-            break;
-        }
-        g_LayoutObjVertical--;
-    }
-}
-
-void CreateEntitiesAbove(s16 arg0) {
-    u8 flag;
-    s32 expected;
-
-    if (g_LayoutObjPosVertical != 0) {
-        FindFirstEntityAbove(arg0 - g_ScrollDeltaY);
-        g_LayoutObjPosVertical = 0;
-    }
-
-    while (true) {
-        if ((g_LayoutObjVertical->posY == 0xFFFF) ||
-            (arg0 < g_LayoutObjVertical->posY)) {
-            return;
-        }
-
-        expected = 0;
-        flag = (g_LayoutObjVertical->entityRoomIndex >> 8) + 0xFF;
-        if ((flag == 0xFF) ||
-            (g_entityDestroyed[flag >> 5] & (1 << (flag & 0x1F))) == expected) {
-            CreateEntityWhenInHorizontalRange(g_LayoutObjVertical);
-        }
-        g_LayoutObjVertical++;
-    }
-}
-
-INCLUDE_ASM("st/cen/nonmatchings/11280", CreateEntitiesBelow);
-
-void InitRoomEntities(s32 objLayoutId) {
-    u16* pObjLayoutStart = g_pStObjLayoutHorizontal[objLayoutId];
-    Tilemap* tilemap = &g_Tilemap;
-    s16 temp_s0;
-    s16 arg0;
-    s16 i;
-    u16* temp_v1;
-
-    g_LayoutObjHorizontal = pObjLayoutStart;
-    g_LayoutObjVertical = g_pStObjLayoutVertical[objLayoutId];
-
-    if (*pObjLayoutStart != 0xFFFE) {
-        g_LayoutObjHorizontal = pObjLayoutStart + 1;
-        arg0 = Random() & 0xFF;
-        for (i = 0; true; i++) {
-            temp_v1 = g_LayoutObjHorizontal;
-            g_LayoutObjHorizontal = temp_v1 + 1;
-            arg0 -= temp_v1[0];
-            if (arg0 < 0) {
-                break;
-            }
-            g_LayoutObjHorizontal = temp_v1 + 3;
-        }
-        g_LayoutObjHorizontal = (temp_v1[2] << 0x10) + temp_v1[1];
-        ((u16*)g_LayoutObjVertical) += i * 2 + 2;
-        g_LayoutObjVertical =
-            (g_LayoutObjVertical->posY << 0x10) + g_LayoutObjVertical->posX;
-    }
-    arg0 = tilemap->scrollX.i.hi;
-    temp_s0 = arg0 + 0x140;
-    i = arg0 - 0x40;
-    if (i < 0) {
-        i = 0;
-    }
-
-    g_LayoutObjPosHorizontal = 0;
-    g_LayoutObjPosVertical = 0;
-    FindFirstEntityToTheRight(i);
-    CreateEntitiesToTheRight(temp_s0);
-    FindFirstEntityAbove(tilemap->scrollY.i.hi + 0x120);
-}
-
-INCLUDE_ASM("st/cen/nonmatchings/11280", UpdateRoomPosition);
-
-void CreateEntityFromCurrentEntity(u16 entityId, Entity* entity) {
-    DestroyEntity(entity);
-    entity->entityId = entityId;
-    entity->pfnUpdate = PfnEntityUpdates[entityId - 1];
-    entity->posX.i.hi = g_CurrentEntity->posX.i.hi;
-    entity->posY.i.hi = g_CurrentEntity->posY.i.hi;
-}
-
-void CreateEntityFromEntity(u16 entityId, Entity* source, Entity* entity) {
-    DestroyEntity(entity);
-    entity->entityId = entityId;
-    entity->pfnUpdate = PfnEntityUpdates[entityId - 1];
-    entity->posX.i.hi = source->posX.i.hi;
-    entity->posY.i.hi = source->posY.i.hi;
-}
-
-#include "../entity_is_near_player.h"
+#include "../libstage.h"
```

Comment:
I can image this being controversial. I chose a single header to make it easier to continue appending as things like `EntityRedDoor` or others are found to match. I don't have a good idea right now how many contiguous functions will be found to be shared, but based on the rest of this file, I'm assuming quite a bit more.